docs/functional_architecture.md

```diff
@@ -0,0 +1,104 @@
+on how to access resources inside a ZIM from a given web resource). This relative URL is hence dependant on the
+location of the resource currently being rewriten.
+
+The table below gives some examples of what the rewritten URL is going to be, depending on the URL of the rewritten document.
+
+|--|--|
+| HTML document URL | image URL rewritten for usage inside the ZIM |
+| `https://en.wikipedia.org/wiki/Kiwix` | `./File:Kiwix_logo_v3.svg` |
+| `https://en.wikipedia.org/wiki` | `./wiki/File:Kiwix_logo_v3.svg` |
+| `https://en.wikipedia.org/waka/Kiwix` | `../wiki/File:Kiwix_logo_v3.svg` |
+| `https://fr.wikipedia.org/wiki/Kiwix` | `../../en.wikipedia.org/wiki/File:Kiwix_logo_v3.svg` |
+
+As can be seen on the last line (but this is true for all URLs), this rewriting has to take into account the convention saying
+ at which ZIM path a given web resource will be stored.
+
+### Dynamic case
+
+The explanation above more or less assumed that the transformations can be done statically, i.e warc2zim can open every
+known document, find existing URLs and replace them with their counterpart inside the ZIM.
+
+While this is possible for HTML and CSS documents typically, it is not possible when the URL is dynamically computed. This is
+typically the case for JS documents, where in the general case the URL is not statically stored inside the JS code but computed
+on-the-fly by aggregating various strings and values.
+
+Rewriting these computations is not deemed feasible due to the huge variety of situation which might be encountered.
+
+A specific function is hence needed to rewitre URL **live in client browser**, intercept any function triggering a web request,
+transform the URL according to conventions (where we expect the resource to be located in the general case) and fuzzy rules.
+
+### Fuzzy rules
+
+The same fuzzy rules that have been used to compute the ZIM path from a resource URL have to be applied again when rewriting URLs.
+
+While this is expected to serve mostly for the dynamic case, we still applies them on both side (staticaly and dynamicaly) for coherency.
+
+### DS rules
+
+**TODO:** why do we need the DS rules in addition to the fuzzy rules, why these rules are applied only statically, and how these
+ transformations are kept in sync with ZIM path computation, i.e. why they are not ran when we compute the ZIM path of a given WARC entry.
+ I don't get how this could work. See https://github.com/openzim/warc2zim/pull/192#issue-2159475902 where it is written "This two fixes come
+ from wabac rewriting but don't seems to be necessary (at least for this fix) and are not tested.".
+
+### Documents considered
+
+For now warc2zim rewrites HTML, CSS, JSON, JSONP and JS documents. Other types of documents are supposed to be either not feasible / not worth it
+(e.g. URLs inside PDF documents), not necessary (e.g. images, fonts) or planned for later due to limited usage in the wild (e.g. XML).
```

Comment:
This is not clear. What would rewriting images mean?

Reply:
Nothing, this is why it is not done ; I changed the wording to try being clearer